### PR TITLE
Adds check to verify sudo

### PR DESCRIPTION
**Pull request by eemperor**

### CODE REVIEW

**Comment by eemperor** [comment]
go codebuild go

**Review by lorengordon — COMMENTED** [review]
Soo, I'm not seeing anywhere in this script or in how it is invoked that would set `-e`... Without that, bash exits with the exit code of the _last_ command. Which means the `watchmaker --version` could fail, and then the sudoers test could succeed, and the test would report success. I think we need to add `set -e`, probably right after the shebang. But if that causes a problem we could set it at the beginning of the "WAM TESTS" section...

File: templates/lx_test.sh [hunk 1]
```diff
@@ -67,6 +67,9 @@ if [ "$build_type" != "$build_type_builder" ] && [ "$${userdata_status[0]}" -eq
   else
     watchmaker --version
   fi
+
+  # Test sudo is functional
+  sudo cat /etc/sudoers
```
Comment (lorengordon) [inline templates/lx_test.sh:72]: Looking at the error in one of the codebuild jobs, I think we can avoid the timeout by using the --non-interactive switch...

```suggestion
  sudo --non-interactive cat /etc/sudoers
```

Comment (lorengordon) [reply templates/lx_test.sh:72]: Oh, sudo has a --list option also, to print out the user's permissions...
```suggestion
  sudo --non-interactive --list
```

Comment (eemperor) [reply templates/lx_test.sh:72]: Okay.  I'll make the change.

Comment (eemperor) [reply templates/lx_test.sh]: I switched to something less verbose.  Hope that's cool also.

**Comment by eemperor** [comment]
Looks like the Codebuild job is running the join-domain state so it won't fail.

**Comment by eemperor** [comment]
go codebuild go

**Comment by lorengordon** [comment]
> Looks like the Codebuild job is running the join-domain state so it won't fail.

Oh dang. Well if you build it in codebuild directly and use the default content, I'll take it.

(Note to self: add a second codebuild integration that uses the default config)

**Comment by eemperor** [comment]
Strange.  It failed this time.

**Comment by eemperor** [comment]
It may be the `--validate` option.  Going to try `--list`.

**Comment by eemperor** [comment]
go codebuild go

**Review by lorengordon — APPROVED** [review]

**Comment by ferricoxide** [comment]
> It may be the `--validate` option. Going to try `--list`.

Might be worth noting that Red Hat, at least up through when I was working on prior contract, recommended the use of `visudo -c`. It's use will turns up more than just "does user have a sudoers entry":
~~~
# visudo -c
/etc/sudoers: parsed OK
/etc/sudoers.d/README: parsed OK
/etc/sudoers.d/sshd_service: bad permissions, should be mode 0440
/etc/sudoers.d/user_maintuser: bad permissions, should be mode 0440
~~~

**Comment by eemperor** [comment]
> > It may be the `--validate` option. Going to try `--list`.
> 
> Might be worth noting that Red Hat, at least up through when I was working on prior contract, recommended the use of `visudo -c`. Its use will turns up more than just "does user have a sudoers entry":
> 
> ```
> # visudo -c
> /etc/sudoers: parsed OK
> /etc/sudoers.d/README: parsed OK
> /etc/sudoers.d/sshd_service: bad permissions, should be mode 0440
> /etc/sudoers.d/user_maintuser: bad permissions, should be mode 0440
> ```

Seems `visudo -c` requires you to be switched to root?  The following happens whether `maintuser` is commented out or not under `/etc/sudoers.d/`.
```
[maintuser@ip-0a0052f6 ~]$ visudo -c
visudo: unable to open /etc/sudoers: Permission denied
```

**Comment by eemperor** [comment]
Test comment

**Pull request merged**